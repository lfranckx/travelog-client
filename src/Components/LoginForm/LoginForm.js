import React, { Component } from 'react';
import AuthApiService from '../../Services/auth-api-service';
import ArticleContext from '../../Contexts/ArticleContext';

export default class LoginForm extends Component {
    static defaultProps = {
        onSubmitSuccess: () => {}
    }

    static contextType = ArticleContext;

    state = { error: null };

    handleSubmitJwtAuth = ev => {
        ev.preventDefault();
        this.setState({ error: null });
        const { username, password } = ev.target;
        AuthApiService.postLogin({
            username: username.value,
            password: password.value,
          })
          .then(res => {
              username.value = '';
              password.value = '';
              this.props.onSubmitSuccess();
          })
          .catch(res => {
              this.setState({ error: res.error });
          });
    }

    render() {
        const { error } = this.state;
        return (
            <form 
                className="auth-form"
                onSubmit={this.handleSubmitJwtAuth}
            >
                <div role='alert'>{error && <p className='error'>{error}</p>}</div>
                <div className="input-box">
                    <label>Username</label>
                    <input
                        type="text" 
                        name='username' 
                        aria-label='username'
                        className='username' 
                        required
                    />
                </div>
                <div className="input-box">
                    <label>Password</label>
                    <input 
                        autoComplete="on"
                        type="password" 
                        name='password' 
                        aria-label='password'
                        className='password'
                        required />
                </div>
                <button type="submit">Login</button>
            </form>
        );
    }
}