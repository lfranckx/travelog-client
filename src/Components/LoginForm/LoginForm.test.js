import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LoginForm from './LoginForm';

describe(`LoginForm component`, () => {
    const props = {
        handleLoginSuccess: () => {}
    };

    it(`renders form given props`, () => {
        const wrapper = shallow(<LoginForm {...props}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    });
})