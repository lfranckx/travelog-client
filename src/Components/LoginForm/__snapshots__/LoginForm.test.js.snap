// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renders form given props 1`] = `
<form
  className="login-form"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="name-input"
  >
    <label>
      Username
    </label>
    <input
      aria-label="username"
      className="username"
      name="username"
      required={true}
      type="text"
    />
  </div>
  <div
    className="name-input"
  >
    <label>
      Password
    </label>
    <input
      aria-label="password"
      autoComplete="on"
      className="password"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <button
    className="submit"
    type="submit"
  >
    Login
  </button>
</form>
`;
